api/integrations/lead_tracking/hubspot/services.py

```diff
@@ -17,13 +18,18 @@ def register_hubspot_tracker(request: Request) -> None:
         )
         return
 
-    logger.info(
-        f"Creating HubspotTracker instance for user {request.user.email} with cookie {hubspot_cookie}"
-    )
-
-    HubspotTracker.objects.update_or_create(
-        user=request.user,
-        defaults={
-            "hubspot_cookie": hubspot_cookie,
-        },
-    )
+    try:
+        HubspotTracker.objects.update_or_create(
+            user=request.user,
+            defaults={
+                "hubspot_cookie": hubspot_cookie,
+            },
+        )
+        logger.info(
+            f"Created HubspotTracker instance for user {request.user.email} with cookie {hubspot_cookie}"
+        )
+    except IntegrityError:
+        logger.info(
+            f"HubspotTracker could not be created for user {request.user.email}"
+            f" due to cookie conflict with cookie {hubspot_cookie}"
+        )
```

Comment:
I'm not sure if I'm missing something here as to why we're using `update_or_create` here, but I don't love the implementation - mostly because `IntegrityError` might not always indicate what we're assuming here. 

I think we should just be more explicit, and do something like: 

```python
if HubspotTracker.objects.filter(hubspot_cookie=hubspot_cookie).exists():
    logger.info(
        f"HubspotTracker could not be created for user {request.user.email}"
        f" due to cookie conflict with cookie {hubspot_cookie}"
    )
else:
    HubspotTracker.objects.create(...)
    logger.info(...)
```

Reply:
Ok I've switched to a filter and exclude for the check. I'm keeping the `update_or_create` here because if a users hubspot cookie is changed we want to include that change in their tracking data.